### PR TITLE
Stop yielding skip value

### LINKED ISSUE #2329: Not yielding skipping value ?
Hi,

[multiple_attributes_iterator](https://github.com/ericproulx/grape/blob/master/lib/grape/validations/multiple_attributes_iterator.rb) and [single_attribute_iterator](https://github.com/ericproulx/grape/blob/master/lib/grape/validations/single_attribute_iterator.rb) are yielding attribute that are eventually skip [here](https://github.com/ericproulx/grape/blob/96e2faf1cb6165eab50209fe045f15784b211d59/lib/grape/validations/validators/multiple_params_base.rb#L12) and [here](https://github.com/ericproulx/grape/blob/96e2faf1cb6165eab50209fe045f15784b211d59/lib/grape/validations/validators/base.rb#L50). Does it make sense to just not yield them and forget the `next if skip_value` ? We would have 1 iteration saved per skip

### PR DESCRIPTION
**Pull request by ericproulx**
This PR will stop yielding `skip` value since iterators are already ignoring `skip` values.

Resolves #2329

### CODE REVIEW

File: spec/grape/validations/single_attribute_iterator_spec.rb [hunk 1]
```diff
@@ -35,20 +35,19 @@
...
+            [params[2], :first, false], [params[2], :second, false]
           )
         end
       end
 
       context 'when missing optional value' do
         let(:params) { [Grape::DSL::Parameters::EmptyOptionalValue, 10] }
 
-        it 'marks params with skipped values' do
+        it 'doest not yield skipped values' do
```
Comment (dblock) [inline spec/grape/validations/single_attribute_iterator_spec.rb:48]: typo: does not

**Review by dblock — COMMENTED** [review]
This is a breaking change, but I fail to see how it can break someone 🤔 How was this a problem for your use-case?

At least we should add a paragraph to UPGRADING.

**Comment by ericproulx** [comment]
> This is a breaking change, but I fail to see how it can break someone 🤔 How was this a problem for your use-case?
> 
> At least we should add a paragraph to UPGRADING.

I'm not sure its a breaking change. There's never been a validation done against a skip value. It's was always returned by base class when validating. I figured that why yield these values when it's actually returned automatically.

**Comment by dblock** [comment]
> I'm not sure its a breaking change. There's never been a validation done against a skip value. It's was always returned by base class when validating. I figured that why yield these values when it's actually returned automatically.

I think you're right. I am going to YOLO it.

**Pull request merged**